### DISCUSSION TITLE
So when are we getting actual performance improvements?

### DISCUSSION THREAD

**Discussion by soph1sticat3d** [ppy/osu#26152] [Q&A]
I love the new client, and I'm looking forward to fully switching to it after the PP is implemented there, but there's just one thing that is going to hold me back unless it gets better, and that is input latency, which for me, an insanely meticulous person, is very important. For example, I could bear with McOsu's disgusting UI, but what makes me choose it over stable and lazer is simply the nonexistent input latency compared to those two clients, and lazer feels the worst out of 3. I've got to use and compare all clients on multiple PCs with all kinds of different hardware (1st-3rd gen Intels+600 series Nvidia GPU; 10th-gen i5+AMD 580; Zen1, Zen3 Ryzens+3060ti, Win 7, 10 11), and it's all the same no matter the setup. Kind of bearable on a tablet since tablets generally have higher input latency, but on a mouse (like five different ones too), it is quite noticeable, especially if you're playing 300+ bpm stuff.

**Comment by peppy** [comment]
For us input latency is lowest on lazer on all test hardware, so this is quite surprising.

Pity you didn't provide logs or screenshots or anything as we can't even begin to guide you in the right direction.

If it's the worst as you report then the game likely isn't running fullscreen properly, so figuring out why that's the case is a starting point.

**Reply by soph1sticat3d** [reply]
Couldn't possibly be a me issue since the numerous setups? But sure, what kind of logs or screenshots would you like to see? Lazer says it "runs in exclusive mode, ensuring lowest input latency" though, so everything should be snappy but it obviously isn't.
